src/configure/configure.ts

```diff
@@ -408,98 +446,75 @@ class Orchestrator {
                 { location: vscode.ProgressLocation.Notification, title: Messages.analyzingRepo },
-                () => templateHelper.analyzeRepoAndListAppropriatePipeline(
+                () => templateHelper.analyzeRepoAndListAppropriatePipeline2(
+                    this.inputs.azureSession,
                     this.inputs.sourceRepository.localPath,
                     this.inputs.sourceRepository.repositoryProvider,
                     repoAnalysisResult,
                     this.inputs.pipelineConfiguration.params[constants.TargetResource])
             );
+            appropriatePipelines = remotePipelines;
         }
 
+        let pipelineMap: Map<string, PipelineTemplate[]> = new Map();
+
+        appropriatePipelines.forEach(element => {
+            if (pipelineMap.has(element.label)) {
+                pipelineMap.get(element.label).push(element);
+            }
+            else {
+                pipelineMap.set(element.label, [element]);
+            }
+        });
+
+        let pipelineLabels = Array.from(pipelineMap.keys());
+
         // TO:DO- Get applicable pipelines for the repo type and azure target type if target already selected
-        if (appropriatePipelines.length > 1) {
-            let selectedOption = await this.controlProvider.showQuickPick(
+        if (pipelineLabels.length > 1) {
+            var selectedOption = await this.controlProvider.showQuickPick(
                 constants.SelectPipelineTemplate,
-                appropriatePipelines.map((pipeline) => { return { label: pipeline.label }; }),
+                pipelineLabels.map((pipeline) => { return { label: pipeline }; }),
                 { placeHolder: Messages.selectPipelineTemplate },
                 TelemetryKeys.PipelineTempateListCount);
-            if (extensionVariables.templateServiceEnabled) {
-                this.inputs.pipelineConfiguration.templateInfo = appropriatePipelines.find((pipeline) => {
-                    return pipeline.label === selectedOption.label;
-                });
-            }
-            else {
-                this.inputs.pipelineConfiguration.template = appropriatePipelines.find((pipeline) => {
-                    return pipeline.label === selectedOption.label;
-                });
-            }
+            //only label gets finalized, template isn't final yet
+            this.inputs.potentialTemplates = pipelineMap.get(selectedOption.label);
         }
         else {
-            if (extensionVariables.templateServiceEnabled) {
-                this.inputs.pipelineConfiguration.templateInfo = appropriatePipelines[0];
-            }
-            else {
-                this.inputs.pipelineConfiguration.template = appropriatePipelines[0];
-            }
+            this.inputs.potentialTemplates = pipelineMap.get(pipelineLabels[0]);
         }
+        this.inputs.pipelineConfiguration.template = this.inputs.potentialTemplates[0];
```

Comment:
Why should this be done here? Should it not be done when we are selecting template based on potential templates?

Reply:
At this place we just actually get to know what will be the label of the template. The local/remote decision hasn't been taken yet.
But since we have to make a decision of AKS or webApp depending upon the template's label, that is why done this.